### PR TITLE
Add mutation to delete (void) question template

### PR DESCRIPTION
**Pull request by maiaht**

### CODE REVIEW

File: backend/api/Services/QuestionTemplateService.cs [hunk 1]
```diff
@@ -94,6 +94,15 @@ Status status
             return newQuestionTemplate;
         }
 
+        public QuestionTemplate Delete(QuestionTemplate questionTemplate)
+        {
+            ReorderQuestionTemplate(questionTemplate);
```
Comment (ErlendHaa) [inline backend/api/Services/QuestionTemplateService.cs:99]: Maybe add a comment here to explain the logic behind the reorder. Its not obvious why its here

File: backend/tests/Services/QuestionTemplateService.cs [hunk 2]
```diff
@@ -96,6 +96,25 @@ public void EditQuestionTemplate()
...
+                text:         Randomize.String(),
+                supportNotes: Randomize.String()
+            );
+            var projectCategory  = new ProjectCategoryService(_context).GetAll().First();
+            questionTemplateToDelete = service.AddToProjectCategory(questionTemplateToDelete.Id, projectCategory.Id);
+
+            var deletedQuestionTemplate = service.Delete(questionTemplateToDelete);
+
+            Assert.True(deletedQuestionTemplate.Status == Status.Voided);
+            Assert.Equal(deletedQuestionTemplate.ProjectCategories, questionTemplateToDelete.ProjectCategories);
```
Comment (ErlendHaa) [inline backend/tests/Services/QuestionTemplateService.cs:115]: Add a test to verify the ordering

File: backend/api/Services/QuestionTemplateService.cs (hunk 1 above)
Comment (maiaht) [reply backend/api/Services/QuestionTemplateService.cs:105]: Fixed

File: backend/tests/Services/QuestionTemplateService.cs (hunk 2 above)
Comment (maiaht) [reply backend/tests/Services/QuestionTemplateService.cs:118]: Fixed

**Review by ReedOnly — APPROVED** [review]

**Pull request merged**